### PR TITLE
new docs for publish

### PR DESCRIPTION
**Pull request by thisisjofrank**

### CODE REVIEW

File: runtime/manual/tools/publish.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# deno publish
+
+*This applies to `deno` v.1.17.0. and above.* **Unstable preview feature**.
```
Comment (bartlomieju) [inline runtime/manual/tools/publish.md:3]:
```suggestion
*This applies to `deno` v1.42.0. and above.* **Unstable preview feature**.
```

**Review by bartlomieju — COMMENTED** [review]
I'll defer to @lucacasonato to review this one

File: runtime/manual/tools/publish.md (hunk 1 above)
Comment (lucacasonato) [reply runtime/manual/tools/publish.md:3]: I don't think we should mark it as "unstable preview"

File: runtime/manual/tools/publish.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+## Description
+
+The `deno publish` command is used to publish a package or workspace to [JSR](https://jsr.io/).
+
+The command will upload the package to the registry and make it available for others to use.
+
+## Package Requirements
+
+Your package must have a `name` and `version` and an `exports` field in its `deno.json` file.
```
Comment (lucacasonato) [inline runtime/manual/tools/publish.md:29]: jsr.json is also ok

File: runtime/manual/tools/publish.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    Skip type-checking. If the value of '--no-check=remote' is supplied,
+    diagnostic errors from remote modules will be ignored
+
+- `-h, --help`
+
+    Print help (see a summary with '-h')
+
+## Examples
+
+- Publish your current workspace
```
Comment (lucacasonato) [inline runtime/manual/tools/publish.md:109]: Add example of dry run

**Review by lucacasonato — COMMENTED** [review]
Also link to https://jsr.io/docs/publishing-packages

File: runtime/manual/tools/publish.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+## Description
+
+The `deno publish` command is used to publish a package or workspace to [JSR](https://jsr.io/).
+
+The command will upload the package to the registry and make it available for others to use.
+
+## Package Requirements
+
+Your package must have a `name` and `version` and an `exports` field in its `deno.json` file.
```
Comment (thisisjofrank) [inline runtime/manual/tools/publish.md:29]:
```suggestion
Your package must have a `name` and `version` and an `exports` field in its `deno.json` or `jsr.json` file.
```

File: runtime/manual/tools/publish.md (hunk 3 above)
Comment (thisisjofrank) [reply runtime/manual/tools/publish.md:109]: I've added it below

**Review by lucacasonato — APPROVED** [review]
LGTM

**Pull request merged**